### ISSUE TITLE
Clarify docs on `function.local()` environment variables

### ISSUE THREAD

**Issue by zsiegel92** [modal-labs/modal-client#1885]
[The docs on `App.function`'s `.local` method](https://modal.com/docs/reference/modal.Function#local) say the following (**emphasis** added):

> Calls the function locally, executing it with the given arguments and returning the execution’s result.
> 
> The function will execute in the same environment as the caller, just like calling the underlying function directly in Python. **In particular, secrets will not be available through environment variables.**

This means "only secrets available in the caller environment will be available through environment variables".

The following example demonstrates this point:

```python
import modal

app = modal.App(
    "test-secrets",
)


@app.function()
async def callee():
    import os
    print(os.environ.get("secret_env_var"))
 
@app.function(
    secrets=[
        modal.Secret.from_dict({"secret_env_var": "secret_value"}),
    ],
)
async def caller():
    await callee.local()

@app.local_entrypoint()
async def main():
    print("`caller.remote()`")
    caller.remote()
```

Maybe updated verbiage can save other users from doing this little check like I did! 🙏

**Comment by erikbern** [comment]
Feel free to submit a PR that updates the text!

**Comment by zsiegel92** [comment]
> Feel free to submit a PR that updates the text!

@erikbern https://github.com/modal-labs/modal-client/pull/1887

**Comment by zsiegel92** [comment]
Big Modal fan btw, keep up the amazing product work!

**Comment by mwaskom** [comment]
Closed with #1887
